docs/src/components/CopyToClipboard.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react'
+import * as copyToClipboard from 'copy-to-clipboard'
+
+export type CopyToClipboardProps = {
+  render: (active, onClick) => React.ReactNode
+  timeout?: number
+  value: string
+}
+
+type CopyToClipboardState = {
+  active: boolean
+}
+
+class CopyToClipboard extends React.Component<CopyToClipboardProps, CopyToClipboardState> {
+  state = {
+    active: false,
+  }
+
+  private timeoutId
+
+  static defaultProps = {
+    timeout: 3000,
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.timeoutId)
+  }
+
+  handleClick = () => {
+    const { timeout, value } = this.props
+
+    this.setState({ active: true })
+    this.timeoutId = setTimeout(() => {
```

Comment:
note that race conditions are introduced here, as previous timeout callback might not be fired when new is activated (i.e. `clearTimeout` is not called for previous value of `timeoutId`)

Reply:
Fixed 👍